### PR TITLE
feat: expose custom labels for control plane machines

### LINKED ISSUE #58: request: provide a way to set labels on the machines created by the controlplane provider
when creating a machine deployment in cluster-api one can set labels on the created machines by specifying them in `.spec.template.metadata`. it would be nice if we could do something similar for the controlplane machines created by the k3s controlplane provider.

it would be nice if controlplane machines could be created as `MachineDeployments` in order to benefit from `MachineHealthChecks`, however that's a separate discussion. given they are now created as `Machines` i propose we add a `MachineLabels map[string]string` optional field to the `KThreesControlPlaneSpec` and set them onto the created machine in `generateMachine()`.

@zawachte if you prefer a different approach let me know, otherwise i'll get started on a PR to do this.

### PR DESCRIPTION
**Pull request by lukebond**
fixes #58 

also, it looks like in [my previous PR](https://github.com/cluster-api-provider-k3s/cluster-api-k3s/pull/54) i forgot to regenerate the manifests. would you prefer me to put that in a separate PR to get it out more quickly as a hotfix?

### CODE REVIEW

**Comment by richardcase** [comment]
It would be good to follow the pattern adopted by KCP as @zawachte mentions: https://github.com/cluster-api-provider-k3s/cluster-api-k3s/issues/58#issuecomment-1756332085

**Comment by lukebond** [comment]
okay i took another pass at it, thanks for the feedback :+1:

File: controlplane/api/v1beta1/kthreescontrolplane_types.go
Comment (zawachte) [inline controlplane/api/v1beta1/kthreescontrolplane_types.go:76]: Should we leave a comment that mentions in the next api version we will move the InfrastructureTemplate field into this struct?

**Review by zawachte — DISMISSED** [review]

**Comment by lukebond** [comment]
This needs re-approval because pushing the comment commit dismissed the previous approval @zawachte

**Review by zawachte — APPROVED** [review]

**Comment by lukebond** [comment]
thanks for the review and merge! not sure if you noticed in the description that it includes manifest generation for the previous PR of mine, because i forgot it. think we should get a hotfix out for it.

**Pull request merged**
